### DISCUSSION TITLE
Items with a publication period set to a date in the future are displayed in the web part.

### DISCUSSION THREAD

**Discussion by chenningwsjp** [microsoft-search/pnp-modern-search#1895] [Q&A] [answered]
**Version used**
Ex: 4.4.1.0

**Inquiry Subject**
Items with a publication period set to a date in the future are displayed in the web part.

**Background and History**
・We are planning to use the "PnP Search Results" web part to display aggregated list items retrieved from multiple lists.
・The following columns are available in the list from which the items were obtained.
　　PublishedDateTime   date and time
　　ExpiredDateTime     date and time

・The following query conditions are set when displaying the "PnP Search Results" Web Part.
　　RefinableDate00 <= today
　　RefinableDate01 >= today
(The crawled properties to which the property names RefinableDate00 and RefinableDate01 map are set to ows_PublishedDateTime and ows_ExpiredDateTime.)

The intent is to only display items whose current time is between the "PublishedDateTime" and "ExpiredDateTime " columns.

**Expected behavior**
Only display items whose current time is between the "PublishedDateTime" and "ExpiredDateTime" columns.
As an example, an item with the "PublishedDateTime" of 3/2 00:00 will be displayed as of today, 3/2.
                       an item with the "PublishedDateTime" of 3/3 00:00 are not displayed as of today, 3/2.

**Describe the bug** (= Difference from expected behavior)
Items for which "PublishedDateTime" is set to a future date (time 00:00) are being displayed in the web part.
As an example, an item with the "PublishedDateTime" set to 3/3 00:00 is being displayed as of today, 3/2.

**To Reproduce**
The item was displayed after setting the "PublishedDateTime" in the future (time 00:00).
→The result is different from the expected behavior.
When I set the "PublishedDateTime" to a future date (time 00:01) for the item , it is not displayed.
→As expected operation

**Questions**
Please provide the settings to achieve the following behavior
When the time is set to 00:00 in the "PublishedDateTime" and "ExpiredDateTime" columns, only items whose current time is between the "PublishedDateTime" and "ExpiredDateTime" columns should be displayed.

**Comment by wobba** [comment, marked as answer]
Not sure if you added spaces or not above.. but the query should be:

```
RefinableDate00<=today RefinableDate01>=today
```

Also the date stored in the index is stored as Z time. If the item is added in say PST which is -9 hours, then the translation of "today" can likely break what you are looking for.

You can try to use query variables instead as we then replace the date in the web part based on your local time adjusted to Z time

```
RefinableDate00<={Today} RefinableDate01>={Today}
```

**Reply by chenningwsjp** [reply]
The problem has been solved according to the method you taught me. Thank you very much!
